golang/internal/grpc/grpc.go

```diff
@@ -275,22 +282,24 @@ func grpcLoop(
 				continue
 			} else {
 				log.Info().Msg("Stream connection is up")
+				health.SetHealthGRPCStatus(true)
 			}
 		}
 
 		command := new(agent.AgentCommand)
 		err = stream.RecvMsg(command)
-		if err == io.EOF {
-			log.Info().Msg("End of stream")
-			grpcConn.Client = nil
-			time.Sleep(appConfig.DefaultTimeout)
-			continue
-		}
 		if err != nil {
-			log.Error().Stack().Err(err).Msg("Cannot receive stream")
 			grpcConn.Client = nil
+			health.SetHealthGRPCStatus(false)
+
+			if err == io.EOF {
+				log.Info().Msg("End of stream")
+			} else {
+				log.Error().Stack().Err(err).Msg("Cannot receive stream")
+				// TODO replace the line above with an error status code check and terminate dagent accordingly
```

Comment:
Do we still need this comment, @nandor-magyar? Is this comment from the `GitLab` repository? If yes, what is blocking us from solving this issue?

Reply:
I don't remember exactly, but if EOF means that the remote connection is closed then agent should exit as well, other errors should be investigated what is possible to get here.

Reply:
I'll create an issue.